packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -57,24 +57,28 @@ export function paginateBlogPosts({
   blogTitle,
   blogDescription,
   postsPerPageOption,
+  blogPaginationActive,
 }: {
   blogPosts: BlogPost[];
   basePageUrl: string;
   blogTitle: string;
   blogDescription: string;
   postsPerPageOption: number | 'ALL';
+  blogPaginationActive: boolean;
 }): BlogPaginated[] {
   const totalCount = blogPosts.length;
   const postsPerPage =
     postsPerPageOption === 'ALL' ? totalCount : postsPerPageOption;
-  const numberOfPages = Math.ceil(totalCount / postsPerPage);
+  const numberOfPages = blogPaginationActive
+    ? Math.ceil(totalCount / postsPerPage)
+    : 1;
```

Comment:
That's where we disagree, cf my comment here: https://github.com/facebook/docusaurus/discussions/8311#discussioncomment-4167986

For me this option should plainly disable the whole blog pagination, and ensure that the `theme.BlogListPage` component never gets renderer. 

Also `postsPerPage` is not an API that feels appropriate to control the number of posts displayed on your landing page